src/DotLiquid/Tags/For.cs

```diff
@@ -107,8 +107,8 @@ public override void Render(Context context, TextWriter result)
                     : Convert.ToInt32(context[_attributes["offset"]])
                 : 0;
 
-            int? limit = _attributes.ContainsKey("limit") ? context[_attributes["limit"]] as int? : null;
-            int? to = (limit != null) ? (int?) (limit.Value + from) : null;
+            long? limit = _attributes.ContainsKey("limit") ? context[_attributes["limit"]] as long? : null;

```

Comment:
here also, support either integer type

Reply:
Actually here convert to 32 bit int only as a for loop with a limit outside the 32 bit range is unreasonable execution cost.